test/units/req.upgrade.test.ts

```diff
@@ -1,24 +1,39 @@
 import { opine } from "../../mod.ts";
-import { expect } from "../deps.ts";
+import { deferred, expect } from "../deps.ts";
 import { describe, it } from "../utils.ts";
 
 describe("req", function () {
   describe(".upgrade", function () {
-    it("should upgrade websocket requests", function (done) {
+    it("should upgrade websocket requests", async function (done) {
       let serverSocket: WebSocket;
 
+      const clientSocketClosedDeferred = deferred();
+      const clientSocketPingedDeferred = deferred();
+      const clientSocketPongedDeferred = deferred();
+      const serverSocketClosedDeferred = deferred();
+      const serverSocketPingedDeferred = deferred();
+      const serverSocketPongedDeferred = deferred();
+
       const handleSocket = (socket: WebSocket) => {
         serverSocket = socket;
 
-        socket.addEventListener("open", () => {
+        serverSocket.addEventListener("close", () => {
+          serverSocketClosedDeferred.resolve();
+        });
+
+        serverSocket.addEventListener("open", () => {
           console.log("[server]: ping");
-          socket.send("ping");
+          serverSocket.send("ping");
+          serverSocketPingedDeferred.resolve();
         });
 
-        socket.addEventListener("message", (e) => {
+        serverSocket.addEventListener("message", (e) => {
           if (e.data === "ping") {
             console.log("[server]: pong");
-            socket.send("pong");
+            serverSocket.send("pong");
+            serverSocketPongedDeferred.resolve();
+          } else if (e.data === "pong") {
```

Comment:
Suspect could be a bug in the Deno WebSocket implementation - despite initiating a close event, a `message` event is able to fire after the close and re-instate the "server idle" timeout resulting in the leaking op/timer error.

The fix here is to wait for said message to arrive and then close the socket. Should closing the socket prevent new messages / in-flight messages from being regarded within the WebSocket event loop...? Not sure, would need to read the spec!